receiver/jaegerreceiver/config.go

```diff
@@ -34,8 +33,19 @@ type RemoteSamplingConfig struct {
 type Protocols struct {
 	GRPC          *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
 	ThriftHTTP    *confighttp.HTTPServerSettings `mapstructure:"thrift_http"`
-	ThriftBinary  *confignet.TCPAddr             `mapstructure:"thrift_binary"`
-	ThriftCompact *confignet.TCPAddr             `mapstructure:"thrift_compact"`
+	ThriftBinary  *ProtocolUDP                   `mapstructure:"thrift_binary"`
```

Comment:
Is Thrift a UDP-based protocol? For some reason I thought it is TCP-based.

Reply:
I believe it's not
https://github.com/open-telemetry/opentelemetry-collector/blob/bf818a241e2fb1d7595eb5a097c1990f2dd735dd/receiver/jaegerreceiver/trace_receiver.go#L104-L105

The `confignet.TCPAddr` structure is used just to store endpoint in this case. Inappropriate place for it I believe

Reply:
Thrift binary and compact are indeed UDP based.